src/Coordination/Changelog.cpp

```diff
@@ -1014,8 +1016,57 @@ void Changelog::writeThread()
 {
     WriteOperation write_operation;
     bool batch_append_ok = true;
-    while (write_operations.pop(write_operation))
+    size_t pending_appends = 0;
+    bool try_batch_flush = false;
+
+    const auto flush_logs = [&](const auto & flush)
+    {
+        LOG_TEST(log, "Flushing {} logs", pending_appends);
+
+        {
+            std::lock_guard writer_lock(writer_mutex);
```

Comment:
Looks suspicious to me. Maybe we should take both locks with `scoped_guard` or have single lock or make last_durable_idx atomic?

Reply:
they are not connected in any way, but `last_durable_idx` can be atomic.

Reply:
actually, it's not atomic because I needed the lock for `durable_idx_cv` and `*flush.failed = true` needs to be protected by it bellow.